pkg/controller/render/render_controller.go

```diff
@@ -500,18 +503,30 @@ func (ctrl *Controller) syncGeneratedMachineConfig(pool *mcfgv1.MachineConfigPoo
 		return err
 	}
 
-	if pool.Status.Configuration.Name == generated.Name {
+	if pool.Spec.Configuration.Name == generated.Name {
 		_, _, err = resourceapply.ApplyMachineConfig(ctrl.client.MachineconfigurationV1(), generated)
 		return err
 	}
 
-	glog.V(2).Infof("Pool %s is now targeting: %s", pool.Name, generated.Name)
-	pool.Status.Configuration.Name = generated.Name
-	pool.Status.Configuration.Source = source
-	_, err = ctrl.client.MachineconfigurationV1().MachineConfigPools().UpdateStatus(pool)
+	oldPool, err := json.Marshal(pool)
+	if err != nil {
+		return err
+	}
+	pool.Spec.Configuration.Name = generated.Name
```

Comment:
you need to:

```
newPool := pool.DeepCopy()
newPool.Spec....
newPool.Spec....
newPoolJSON, err := ...
```

Or we mutate informer cache?

Reply:
Yeah I checked for that, AFAICS we copy the pool earlier in `syncMachineConfigPool`.  Mumble mumble Rust `&` vs `&mut` :wink:

Reply:
(I ended up deep copying anyways, since why not)